config/initializers/devise.rb

```diff
@@ -128,6 +128,13 @@
   # When invite_for is 0 (the default), the invitation won't expire.
   config.invite_for = 4.weeks
 
+  # replace the token generator normally instantiate here : https://github.com/heartcombo/devise/blob/88724e10adaf9ffd1d8dbfbaadda2b9d40de756a/lib/devise/rails.rb#L41
+  Devise.token_generator =
+    if (secret_key = Devise.secret_key)
+      CustomDeviseTokenGenerator.new(
+        ActiveSupport::CachingKeyGenerator.new(ActiveSupport::KeyGenerator.new(secret_key))
+      )
+    end
```

Comment:
Je suis partagé entre copier-coller le code original et l’inutilité de la variable intermédiaire `secret_key`, qui plus est avec une affectation dans un `if`, qu’on peut prendre pour un test d’égalité.

```suggestion
  Devise.token_generator = CustomDeviseTokenGenerator.new(ActiveSupport::CachingKeyGenerator.new(ActiveSupport::KeyGenerator.new(Devise.secret_key))
```

?

Reply:
Ouais pareil, j'ai pas mal hésité mais en vrai ta solution me plaît mieux.